File: quant-center-publish/frontend/src/types/marketBriefing.ts
export interface MarketBriefingMetric {
  name: string;
  value: string;
}

export interface MarketBriefingHighlight {
  dimension: string;
  state: string;
  explanation: string;
}

export interface MarketBriefingDimension {
  name: string;
  state: string;
  explanation: string;
  metrics: MarketBriefingMetric[];
}

export interface MarketBriefingSnapshot {
  scope: string;
  approach: 'Aggressive' | 'Normal' | 'Cautious' | 'Defensive';
  confidence: 'High' | 'Medium' | 'Low';
  oneLineSummary: string;
  reason: string;
  keyPositives: MarketBriefingHighlight[];
  keyRisks: MarketBriefingHighlight[];
  dimensions: MarketBriefingDimension[];
  rawMetrics: Record<string, unknown>;
  dataDate: string;
  universeSize: number;
  sectorCoverage: number;
  lastRefreshTime: string;
}

File: quant-center-publish/backend/app/schemas/market_briefing.py
"""Read-only API contract for the existing Market Intelligence outputs."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class MarketBriefingMetric(BaseModel):
    name: str
    value: str


class MarketBriefingHighlight(BaseModel):
    dimension: str
    state: str
    explanation: str


class MarketBriefingDimension(BaseModel):
    name: str
    state: str
    explanation: str
    metrics: list[MarketBriefingMetric] = Field(default_factory=list)


class MarketBriefingSnapshot(BaseModel):
    scope: str
    approach: str
    confidence: str
    oneLineSummary: str
    reason: str
    keyPositives: list[MarketBriefingHighlight] = Field(default_factory=list)
    keyRisks: list[MarketBriefingHighlight] = Field(default_factory=list)
    dimensions: list[MarketBriefingDimension] = Field(default_factory=list)
    rawMetrics: dict[str, Any] = Field(default_factory=dict)
    dataDate: str
    universeSize: int
    sectorCoverage: int
    lastRefreshTime: str
